Update using Flutter from China (#9338)
flutter/website: src/community/china/_os-settings.md

{% assign id =  include.os | downcase -%}

{% if id == 'windows' -%}
{% assign shell = 'Powershell' -%}
{% assign prompt = 'C:\>' -%}
{% assign envvarset = 'C:\> $env:' -%}
{% assign setpath = 'C:\> $env:PATH = $pwd.PATH + "/flutter/bin",$env:PATH -join ";"' -%}

> {% assign comtoset = 'env:' -%}

Reviewer: ```suggestion
{% assign comtoset = '$env:' -%}
```

> {% assign unzip='Extract-Archive' %}

Reviewer 2: To be consistent with the other commands and fix syntax highlighting, should this start with a `C:\>`?

```suggestion
{% assign unzip='C:\> Extract-Archive' %}
```

{% capture permaddexample -%}
$newPath = $pwd.PATH + "/flutter/bin",$env:PATH -join ";"
[System.Environment]::SetEnvironmentVariable('Path',$newPath,User)
[System.Environment]::SetEnvironmentVariable('PUB_HOSTED_URL','https://pub.flutter-io.cn',User)
[System.Environment]::SetEnvironmentVariable('FLUTTER_STORAGE_BASE_URL','https://storage.flutter-io.cn',User)
{% endcapture -%}
{% else -%}
{% assign shell = 'your terminal' -%}
{% assign prompt = '$' -%}
{% assign envvarset = '$ export ' -%}
{% assign setpath = '$ export PATH="$PWD/flutter/bin:$PATH"' -%}
{% assign comtoset = 'export' -%}

> {% assign unzip='unzip' %}

Reviewer 2: To be consistent with the other commands and fix syntax highlighting, should this start with a `$`?

```suggestion
{% assign unzip='$ unzip' %}
```

{% capture permaddexample -%}
cat <<EOT >> ~/.zprofile
{{envvarset}}PUB_HOSTED_URL="https://pub.flutter-io.cn"
{{envvarset}}FLUTTER_STORAGE_BASE_URL="https://storage.flutter-io.cn"
{{setpath}}
EOT
{% endcapture -%}
{% endif -%}

<div id="{{id}}" class="tab-pane
  {%- if id == 'windows' %} active {% endif %}"
  role="tabpanel" aria-labelledby="{{id}}-tab" markdown="1">

> 1. Create and change into the directory where you plan
>    to store your local Flutter clone.

Reviewer: ```suggestion
1. Open a new Terminal window to prepare running scripts.
```

> 1. Set your pub URL to your mirror site.

Reviewer: ```suggestion
1. Set `PUB_HOSTED_URL` to your mirror site.
```

   In {{shell}}, run the following command:

   ```terminal
   {{envvarset}}PUB_HOSTED_URL="https://pub.flutter-io.cn"
   ```

> 1. Set your Flutter storage URL to your mirror site.

Reviewer: ```suggestion
1. Set `FLUTTER_STORAGE_BASE_URL` to your mirror site.
```

   ```terminal
   {{envvarset}}FLUTTER_STORAGE_BASE_URL="https://storage.flutter-io.cn"
   ```

> 1. Download the Flutter archive from your mirror site.

Reviewer 2: "Archive" implies the entire archive, perhaps "build" or "installation bundle".

```suggestion
1. Download your the Flutter installation bundle for
   your platform from your mirror site.
```

Author: A zip file or a tar file are archive file formats. Going to keep as-is.

Reviewer 2: That's technically true, but it's quite confusing since it uses "archive" to mean two different things in this same step. It reads as if it's telling me to download the entire archive page.

> 1. Download the Flutter archive from your mirror site.
>    In your preferred browser, go to https://flutter.cn/docs/release/archive.

Reviewer: Not sure if it's working as intended.

```suggestion
1. Download the Flutter archive from your mirror site.
   In your preferred browser, go to
   [Flutter SDK archive](https://flutter.cn/docs/release/archive?tab={{id}}).
```

>    [Flutter SDK archive](https://flutter.cn/docs/release/archive?tab={{id}}).

Reviewer 2: These instruction will be incorrect for most other mirror sites as they don't host a web version of our archive like CFUG does.

Perhaps you can provide two paths. One for the CFUG mirror and link to their archive, "Otherwise, follow the instructions in [Download Flutter archives based on a mirrored URL](#download-flutter-archives-based-on-a-mirrored-url)" or similar.

Author: This entire procedure starts with "All examples that follow presume that you are using the CFUG mirror." Keeping as-is.

1. Create a folder where you can install Flutter.

   Consider
   {% if id == 'windows' -%}
   `%USERPROFILE%` or `D:\dev`.
   {% else -%}
   `~/dev`.
   {% endif %}

1. Extract the zip file.

   ```terminal
   {{unzip}} flutter_sdk_v1.0.0.zip

>    ```

Reviewer 2: I think we should be clear here that this command won't just work, and they need to replace the filename with the one they downloaded. A full example can always be provided too if desired.

We also often use syntax like `<downloaded zip name>` or similar to indicate a required CLI argument.

1. Add Flutter to your `PATH` environment variable.
   In {{shell}}, run the following command:

   ```terminal
   {{setpath}}
   ```

1. Run Flutter Doctor to verify your installation.
   In {{shell}}, run the following command:

   ```terminal
   {{prompt}} flutter doctor
   ```

1. Return to the [setting up Flutter]({{site.url}}/get-started/editor)
   guide and continue from that procedure.

> From this example, `flutter pub get` fetches packages from `flutter-io.cn`.
> in any terminal where you set `PUB_HOSTED_URL` and `FLUTTER_STORAGE_BASE_URL`.

Reviewer: ```suggestion
From this example, `flutter pub get` fetches packages from `flutter-io.cn`,
in any terminal where you set `PUB_HOSTED_URL` and `FLUTTER_STORAGE_BASE_URL`.
```

   {% else %}
      {% capture unzip -%} {{prompt}} tar -xf{% endcapture -%}
   {% endif %}
   {% capture permaddexample -%}
cat <<EOT >> ~/.zprofile
{{envvarset}}PUB_HOSTED_URL="https://pub.flutter-io.cn"
{{envvarset}}FLUTTER_STORAGE_BASE_URL="https://storage.flutter-io.cn"
{{setpath}}
EOT
   {% endcapture -%}
{% endif -%}
{%- case id %}
   {% when 'windows','macos' %}
      {%- assign file-format = 'zip' %}
      {%- assign download-os = id %}
   {% when 'linux','chromeos' %}
      {%- assign download-os = 'linux' %}
      {%- assign file-format = 'tar.xz' %}
{% endcase %}

<div id="{{id}}" class="tab-pane
  {%- if id == 'windows' %} active {% endif %}"
  role="tabpanel" aria-labelledby="{{id}}-tab" markdown="1">

This procedure requires using {{shell}}.

1. Open a new window in {{shell}} to prepare running scripts.

1. Set `PUB_HOSTED_URL` to your mirror site.

   ```terminal
   {{envvarset}}PUB_HOSTED_URL="https://pub.flutter-io.cn"
   ```

1. Set `FLUTTER_STORAGE_BASE_URL` to your mirror site.

   ```terminal
   {{envvarset}}FLUTTER_STORAGE_BASE_URL="https://storage.flutter-io.cn"
   ```

1. Download the Flutter archive from your mirror site.

>    In your preferred browser, go to

Reviewer 2: ```suggestion
   In your preferred browser, go to the
```

   [Flutter SDK archive](https://flutter.cn/docs/release/archive?tab={{id}}).

> 1. Create a folder where you can install Flutter. then change into it.

Reviewer 2: ```suggestion
1. Create a folder where you can install Flutter, then change into it.
```

   Consider a path like {{installdirsuggestion}}.

   ```terminal
   {{newdir}}; cd {{installdirsuggestion}}
   ```

> 1. Extract the the SDK from the {{file-format}} archive file.

Reviewer 2: ```suggestion
1. Extract the the SDK from the `{{file-format}}` archive file.
```

   This example assumes you downloaded the {{os}} version of the Flutter SDK.

   ```terminal
   {{unzip}} {{sdk | replace: "opsys", download-os}}{{file-format}}

>    ```

Reviewer 2: I recommend adding an extra sentence to explicitly explain that they need to use the exact name of the specific bundle they downloaded here. When there are terminal code blocks, readers often just copy paste without much consideration, or they may be new to this type of activity and may not know.

1. Add Flutter to your `PATH` environment variable.

   ```terminal
   {{setpath}}
   ```

1. Run Flutter Doctor to verify your installation.

   ```terminal
   {{prompt}} flutter doctor
   ```

1. Return to the [setting up Flutter]({{site.url}}/get-started/editor)
   guide and continue from that procedure.

From this example, `flutter pub get` fetches packages from `flutter-io.cn`,

> in any terminal where you set `PUB_HOSTED_URL` and `FLUTTER_STORAGE_BASE_URL`.

Reviewer 2: What do you intend to mean by "from this example"? "After completing the steps of this procedure"?

Any environment variables set using `{{comtoset}}` in this procedure
only apply to the current window.

To set these values on a permanent basis,
{% if id == 'windows' -%}
set the enviroment variables as in the following example:
{% else -%}

> append those three `export` commands to the `*rc` or `*profile`

Reviewer 2: It's been a while since the `export` commands at this point. I wasn't sure what you were referring to at first.

```suggestion
append the previous three `export` commands to the `*rc` or `*profile`
```
